### PR TITLE
cider: init at 1.3.1308

### LINKED ISSUE #162683: Packaging request: Cider
**Project description**
Cider is an open-source and community-oriented Apple Music client.

**Metadata**

* homepage URL: https://cider.sh/
* source URL: https://github.com/ciderapp/Cider
* license: GNU Affero General Public License v3.0
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by cigrainger**
###### Description of changes

Adds [Cider](https://cider.sh), an open-source Apple Music client. Fixes https://github.com/NixOS/nixpkgs/issues/162683.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/cider/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ appimageTools, lib, fetchurl }:
+
+appimageTools.wrapType2 rec {
+  pname = "cider";
+  version = "1.3.1308";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cider/default.nix:6]:
```suggestion
```
no longer required

File: pkgs/applications/audio/cider/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      install -m 444 -D ${contents}/${pname}.desktop -t $out/share/applications
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
+      cp -r ${contents}/usr/share/icons $out/share
+    '';
+
+  meta = with lib; {
+    description =
+      "A new look into listening and enjoying Apple Music in style and performance.";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cider/default.nix:27]:
```suggestion
    description = "A new look into listening and enjoying Apple Music in style and performance.";
```

File: pkgs/applications/audio/cider/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ appimageTools, lib, fetchurl }:
+
+appimageTools.wrapType2 rec {
+  pname = "cider";
+  version = "1.3.1308";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url =
+      "https://1308-429851205-gh.circle-artifacts.com/0/%7E/Cider/dist/artifacts/Cider-${version}.AppImage";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cider/default.nix:10]:
```suggestion
    url = "https://1308-429851205-gh.circle-artifacts.com/0/%7E/Cider/dist/artifacts/Cider-${version}.AppImage";
```
You are sure this URL stays similar?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/cider/default.nix (hunk 3 above)
Comment (cigrainger) [reply pkgs/applications/audio/cider/default.nix:9]: I'm not sure, no. But it has thus far.

File: pkgs/applications/audio/cider/default.nix (hunk 1 above)
Comment (cigrainger) [reply pkgs/applications/audio/cider/default.nix:6]: I originally committed this but then reverted it as it caused tests to fail.

Comment (SuperSandro2000) [reply pkgs/applications/audio/cider/default.nix:6]: You mean line 15? That also accepts pname+version.

Comment (cigrainger) [reply pkgs/applications/audio/cider/default.nix:6]: Nope I switched that too (line 16) and tests still failed.

Comment (SuperSandro2000) [reply pkgs/applications/audio/cider/default.nix:6]: How do they fail? I need some more information to better help you.

Comment (cigrainger) [reply pkgs/applications/audio/cider/default.nix]: @SuperSandro2000 I've just removed it again so you can see in the checks. The basic eval check fails.

File: pkgs/applications/audio/cider/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  version = "1.3.1308";
+
+  src = fetchurl {
+    url = "https://1308-429851205-gh.circle-artifacts.com/0/%7E/Cider/dist/artifacts/Cider-${version}.AppImage";
+    sha256 = "1lbyvn1c8155p039qfzx7jwad7km073phkmrzjm0w3ahdpwz3wgi";
+  };
+
+  extraInstallCommands =
+    in ''
+    let contents = appimageTools.extract { inherit src; };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cider/default.nix:14]:
```suggestion
  extraInstallCommands = let
    contents = appimageTools.extract { inherit src; };
  in ''
```

Comment (cigrainger) [reply pkgs/applications/audio/cider/default.nix]: :facepalm: Sorry about that. Thanks for the catch.

**Pull request merged**